### DISCUSSION TITLE
Debugging Otel Items

### DISCUSSION THREAD

**Discussion by drusellers** [open-telemetry/opentelemetry-dotnet#4816] [Q&A]
- Otel Libraries
    - OpenTelemetry.Extensions.Hosting 1.5.1
    - OpenTelemetry.Exporter.OpenTelemetryProtocol 1.5.1
- DotNet 7

Setting OTEL debug output for dotnet

```sh
  export OTEL_LOG_LEVEL=debug
  export OTEL_DOTNET_AUTO_LOG_DIRECTORY=$(pwd)/otel
  mkdir -p $OTEL_DOTNET_AUTO_LOG_DIRECTORY
  export OTEL_DOTNET_AUTO_TRACES_CONSOLE_EXPORTER_ENABLED=true
dotnet run --project src/Agent
```

This doesn't end up writing any content to the filesystem. Am I missing something?

Also, while trying to debug traces with 

```csharp
                    builder.AddOtlpExporter(o =>
                    {
                        o.Protocol = OtlpExportProtocol.HttpProtobuf;
                        o.BatchExportProcessorOptions = new BatchExportActivityProcessorOptions
                        {
                            MaxQueueSize = 2048,
                            ScheduledDelayMilliseconds = 5000,
                            ExporterTimeoutMilliseconds = 30000,
                            MaxExportBatchSize = 512
                        };
                        o.ExportProcessorType = ExportProcessorType.Batch;
                    });
```

I don't see anything in the HTTP Logs for DotNet or anything in my HTTP proxy.

```
log.MinimumLevel.Override("System.Net.Http.HttpClient", LogEventLevel.Debug);
```

But I am getting data in Tempo. Any thing else I can be configuring to see the HTTP request traffic?

**Comment by Kielek** [comment]
`OTEL_LOG_LEVEL`, `OTEL_DOTNET_AUTO_LOG_DIRECTORY`, `OTEL_DOTNET_AUTO_TRACES_CONSOLE_EXPORTER_ENABLED` are supported by https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/ and cannot be used directly with OpenTelemetry API/SDK which you are using.

Some diagnostics can be enabled by https://github.com/open-telemetry/opentelemetry-dotnet/blob/f3bc6a2d9aacc5964f5a04398b8269546dba81a2/src/OpenTelemetry/README.md?plain=1#L95-L96

**Reply by drusellers** [reply]
@Kielek ah, ok. TY for clarifying the issues around the ENV VAR's that I was trying to use. :) 

Do you know why I'm not seeing any calls in the HTTP Client logs? I'd expect to see data being sent to my instance of Tempo.
